src/http/cookie.cr

```diff
@@ -140,6 +156,40 @@ module HTTP
       end
     end
 
+    # Returns `false` if `#name` has a security prefix but the requirements are not met as per
+    # [RFC 6265 bis §4.1.3](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-07#section-4.1.3),
+    # otherwise returns `true`.
+    def valid? : Bool
+      self.valid_secure_prefix? && self.valid_host_prefix?
+    end
+
+    # Raises `ArgumentError` if `self` is not `#valid?`.
+    def validate! : self
+      raise ArgumentError.new "Invalid cookie name. Has '__Secure-' prefix, but is not secure." unless self.valid_secure_prefix?
+      raise ArgumentError.new "Invalid cookie name. Does not meet '__Host-' prefix requirements of: secure: true, path: \"/\", domain: nil." unless self.valid_host_prefix?
+
+      self
+    end
+
+    private def valid_secure_prefix? : Bool
+      self.secure || !@name.starts_with?("__Secure-")
```

Comment:
```suggestion
      self.secure || (@secure.nil? && !@name.starts_with?("__Secure-"))
```

Reply:
Hm, yeah but shouldn't it be `@secure != false || !@name.starts_with?("__Secure-")`?

Reply:
Is this scenario even possible? I'm struggling to think about how it would get into that state.

* Can't pass `nil` to `secure=`
* If you use an explicit `nil` in the constructor w/o the prefix, its valid
* If you use an explicit `nil` in the constructor with the prefix, it gets set to `true`
* If you set the name after creating it with no or `nil` value, it still gets set the `true`

Reply:
Yeah, I can't think of how to reach that state. But I suppose it still wouldn't hurt to cover this case.

Reply:
> Hm, yeah but shouldn't it be `@secure != false || !@name.starts_with?("__Secure-")`?

@straight-shoota I don't think so, since `nil != false # => true`

Reply:
Mmm good call. I did think having both `be_true` in the next test case was kinda weird. I went back and played with it more. I think it should actually be `self.secure || !@name.starts_with?("__Secure-")`. This makes all tests pass, after changing the one to `be_false` after setting `@secure` to `nil`.

The cookie would always be valid if it's true, otherwise it's only valid if it doesnt have the prefix.

Reply:
The most correct is IMO `self.secure || (@secure.nil? && !@name.starts_with?("__Secure-"))`.

Reply:
That causes a bunch of spec failures, such as causing `HTTP::Cookie.new "x", "", secure: false` to raise saying it has the prefix but isn't secure. Which isn't correct.